Game/model/model.gd

```diff
@@ -282,6 +286,49 @@ func get_storage_cap(player_id: int, type: Types.Item) -> float:
 		return 0.0
 
 
+## Set the heat data for the given player to the data given. Is an RPC.
+@rpc("any_peer", "call_local", "reliable")
+func add_heat_data_at(
+		player_id: int, position: Vector2i, heat: float, heat_capacity: float
+	) -> void:
+	# start function
+	var player_state: PlayerState = player_states.get_state(player_id)
+	var heat_data: HeatData = HeatData.new(position, heat, heat_capacity)
+	player_state.heat_data_list.append(heat_data)
+	heat_data_updated.emit()
+
+
+## Delete the heat data for the given player at the given position. Is an RPC.
+@rpc("any_peer", "call_local", "reliable")
+func remove_heat_data_at(player_id: int, position: Vector2i) -> void:
+	var player_state: PlayerState = player_states.get_state(player_id)
+
+	var heat_data_list: Array[HeatData] = player_state.heat_data_list
+	var index_to_remove: int = heat_data_list.find_custom(
+		func(elem): return elem.position == position
+	)
+
+	if index_to_remove != -1:
+		heat_data_list.remove_at(index_to_remove)
+		heat_data_updated.emit()
+
+
+## Set the heat data heat value at the given position to the given value. Is an RPC.
+@rpc("any_peer", "call_local", "reliable")
+func set_heat_to(player_id: int, position: Vector2i, new_heat: float) -> void:
```

Comment:
Make sure we're not calling this for every cell. We probably want a single rollup call setting all heat once per frame at most.